php-plugin/src/main/java/fr/greencodeinitiative/php/PhpRuleRepository.java

```diff
@@ -0,0 +1,119 @@
+ */
+package fr.greencodeinitiative.php;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableList;
+import fr.greencodeinitiative.php.checks.AvoidDoubleQuoteCheck;
+import fr.greencodeinitiative.php.checks.AvoidFullSQLRequestCheck;
+import fr.greencodeinitiative.php.checks.AvoidSQLRequestInLoopCheck;
+import fr.greencodeinitiative.php.checks.AvoidTryCatchFinallyCheck_NOK_failsAllTryStatements;
+import fr.greencodeinitiative.php.checks.AvoidUsingGlobalVariablesCheck;
+import fr.greencodeinitiative.php.checks.IncrementCheck;
+import fr.greencodeinitiative.php.checks.NoFunctionCallWhenDeclaringForLoop;
+import fr.greencodeinitiative.php.checks.UseOfMethodsForBasicOperations;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.api.server.rule.RulesDefinitionAnnotationLoader;
+import org.sonar.plugins.php.api.visitors.PHPCustomRuleRepository;
+
+public class PhpRuleRepository implements RulesDefinition, PHPCustomRuleRepository {
+
+  public static final String LANGUAGE = "php";
+  public static final String NAME = "Green Code Initiative";
+  public static final String RESOURCE_BASE_PATH = "/fr/greencodeinitiative/l10n/php/rules/custom/";
+  public static final String REPOSITORY_KEY = "gci-php";
+
+  @Override
+  public void define(Context context) {
+    NewRepository repository = context.createRepository(repositoryKey(), LANGUAGE).setName(NAME);
+
+    new RulesDefinitionAnnotationLoader().load(repository, checkClasses().toArray(new Class[] {}));
+
+    // technical debt
+    Map<String, String> remediationCosts = new HashMap<>();
+    remediationCosts.put(AvoidSQLRequestInLoopCheck.RULE_KEY, "10min");
+    remediationCosts.put(AvoidFullSQLRequestCheck.RULE_KEY, "20min");
+    repository.rules().forEach(rule -> {
+      String debt = remediationCosts.get(rule.key());
+
+      // TODO DDC : create support to use org.apache.commons.lang.StringUtils
+//      if (StringUtils.isBlank(debt)) {
+      if (debt == null || debt.trim().equals("")) {
+        // default debt to 5min for issue correction
+        rule.setDebtRemediationFunction(
+                rule.debtRemediationFunctions().constantPerIssue("5min"));
+      } else {
+        rule.setDebtRemediationFunction(
+                rule.debtRemediationFunctions().constantPerIssue(debt));
+      }
+    });
+
+    // HTML description
+    repository.rules().forEach(rule ->
+            rule.setHtmlDescription(loadResource(RESOURCE_BASE_PATH + rule.key() + ".html")));
+
+    repository.done();
+  }
+
+  @Override
+  public String repositoryKey() {
+    return REPOSITORY_KEY;
+  }
+
+  @Override
+  public List<Class<?>> checkClasses() {
+    return ImmutableList.of(
```

Comment:
There is already a list with rules in [RulesList.java#L47](https://github.com/green-code-initiative/ecoCode/blob/7c5f8724b0d7ae690bfc2341b5cd2df9057e96dd/php-plugin/src/main/java/fr/greencodeinitiative/php/RulesList.java#L47), perhaps it should be merged with this one?

Reply:
Like I said below, this is a quick solution ... but having a special class for list rules, I'm thinking about it if it's a good idea. But we will discuss about it after the hackathon, please.

Reply:
np I understand 👍